tests/runtests.bats

```diff
@@ -223,3 +223,64 @@ cleanup() {
 
   cleanup
 }
+
+@test "bring up and save a snapshot" {
+  export VAGRANT_CWD=tests/snapshot
+
+  cleanup
+  run ${VAGRANT_CMD} up ${VAGRANT_OPT}
+  echo "${output}"
+  echo "status = ${status}"
+  [ "$status" -eq 0 ]
+  run ${VAGRANT_CMD} snapshot save default test
+  echo "${output}"
+  echo "status = ${status}"
+  [ "$status" -eq 0 ]
+  run ${VAGRANT_CMD} snapshot list default 
+  echo "${output}"
+  echo "status = ${status}"
+  [ "$status" -eq 0 ]
+  run ${VAGRANT_CMD} snapshot delete default test
+  echo "${output}"
+  echo "status = ${status}"
+  [ "$status" -eq 0 ]
+
+  cleanup
+}
+
+@test "bring up and save a snapshot and restore it" {
+  export VAGRANT_CWD=tests/snapshot
+
+  cleanup
+  run ${VAGRANT_CMD} up ${VAGRANT_OPT}
+  echo "${output}"
+  echo "status = ${status}"
+  [ "$status" -eq 0 ]
+  run ${VAGRANT_CMD} snapshot save default test
```

Comment:
Might need to add another file of a different name before snapshot save here, then save, then delete and verify the file is gone, then after adding the file below that we're looking to be removed by restoring, also want to check that the file that previously existed and was deleted is restored.

Ideally this would ensure that the restore is actually restoring the snapshot that contained previous files and not just rolling back to the original boot state.

It might not work with the tinycore image as I've just discovered it doesn't persist changes if you reboot.